flow/util/distributed.py

```diff
@@ -205,6 +205,23 @@ def read(path):
 
     def read_sweep(this):
         return [*this['minmax'], this['step']]
+    
+    def apply_condition(config, data):
+        # TODO: tune.sample_from only supports 'random' search algorithm. 
+        # To make conditional paraeter for the other algorithms, different 
+        # algorithms should take different methods.
+        if args.algorithm == 'random':
+            gppad = config['CELL_PAD_IN_SITES_GLOBAL_PLACEMENT']
+            config['CELL_PAD_IN_SITES_DETAIL_PLACEMENT']
+            dppad_min, dppad_max = data['CELL_PAD_IN_SITES_DETAIL_PLACEMENT']['minmax']
+            dppad_step = data['CELL_PAD_IN_SITES_DETAIL_PLACEMENT']['step']
+            if dppad_step==1:
+                config['CELL_PAD_IN_SITES_DETAIL_PLACEMENT']= tune.sample_from(lambda spec: tune.randint(dppad_min, spec.config.CELL_PAD_IN_SITES_GLOBAL_PLACEMENT+1))
+            if dppad_step>1:
+                config['CELL_PAD_IN_SITES_DETAIL_PLACEMENT']= tune.sample_from(lambda spec: tune.choice(np.adarray.tolist(np.arange(dppad_min, spec.config.CELL_PAD_IN_SITES_GLOBAL_PLACEMENT+1, dppad_step))))
```

Comment:
`gppad` -> `gp_pad` same for `dp*`, easier to read.
Line 215 looks to be missing some sort of assignment.
Please split lines 219 and 221 in multiple statements, they are too long.
Also run autopep8 for style, e.g., `dppad_step == 1`...

Reply:
Thank you for your comment and sorry for my mistake for redundant lines. I just pushed.